test/server_test.lua

```diff
@@ -152,3 +150,41 @@ g.test_inherit = function()
     local instance = child:new({command = 'test-cmd', workdir = 'test-dir'})
     t.assert_equals(instance.start, Server.start)
 end
+
+g.before_test('test_socket_as_port', function()
+    server:stop()
+    g.server_snap = table.deepcopy(server)
+    fio.rmtree(datadir)
+    fio.mktree(fio.pathjoin(datadir, 'common'))
```

Comment:
I don't like these activities. What are they for?

Reply:
Clean all snap, vylog, xlog files and so on.

Reply:
But why? The test is about the listening socket.

Reply:
In my mind, there are situations when the server would be restarted, read old files, and fails on some wrong. Is not possible?

Reply:
Also, I can change `server_instance.lua`:

`--- a/test/server_instance.lua
+++ b/test/server_instance.lua
@@ -11,7 +11,7 @@ box.cfg({
     listen = listen
 })
 
-box.schema.user.grant('guest', 'read,write,execute', 'universe')
+box.schema.user.grant('guest', 'read,write,execute', 'universe', nil,{if_not_exists=true})
 
 local httpd = require('http.server').new('0.0.0.0', http_port)
`

We need it due to server restart.

Reply:
Yea, it's would be much better